src/Microsoft.SqlTools.ServiceLayer/ModelManagement/ModelOperations.cs

```diff
@@ -0,0 +1,416 @@
+		}
+
+		/// <summary>
+		/// Deletes a model from the given table 
+		/// </summary>
+		/// <param name="connection">Db connection</param>
+		/// <param name="request">model request</param>
+		public virtual void DeleteModel(IDbConnection connection, DeleteModelRequestParams request)
+		{
+			WithDbChange(connection, request, (request) =>
+			{
+				Dictionary<string, object> parameters = new Dictionary<string, object>();
+				using (IDbCommand command = connection.CreateCommand())
+				{
+					command.CommandText = GetDeleteModelQuery(request.TableName, request.SchemaName, request.ModelId, parameters);
+
+					foreach (var item in parameters)
+					{
+						var parameter = command.CreateParameter();
+						parameter.ParameterName = item.Key;
+						parameter.Value = item.Value;
+						command.Parameters.Add(parameter);
+					}
+					command.ExecuteNonQuery();
+					return true;
+				}
+			});
+		}
+
+		/// <summary>
+		/// Configures model table
+		/// </summary>
+		/// <param name="connection">Db connection</param>
+		/// <param name="request">model request</param>
+		public virtual void ConfigureImportTable(IDbConnection connection, ModelRequestBase request)
+		{
+			WithDbChange(connection, request, (request) =>
+			{
+				Dictionary<string, object> parameters = new Dictionary<string, object>();
+				using (IDbCommand command = connection.CreateCommand())
+				{
+					command.CommandText = GetCreateModelTableQuery(request.TableName, request.SchemaName);
+
+					foreach (var item in parameters)
+					{
+						var parameter = command.CreateParameter();
+						parameter.ParameterName = item.Key;
+						parameter.Value = item.Value;
+						command.Parameters.Add(parameter);
+					}
+					command.ExecuteNonQuery();
+					return true;
+				}
+			});
+		}
+
+		/// <summary>
+		/// Verifies model table 
+		/// </summary>
+		/// <param name="connection">Db connection</param>
+		/// <param name="request">model request</param>
+		public virtual bool VerifyImportTable(IDbConnection connection, ModelRequestBase request)
+		{
+			int result = WithDbChange(connection, request, (request) =>
+			{
+				Dictionary<string, object> parameters = new Dictionary<string, object>();
+				using (IDbCommand command = connection.CreateCommand())
+				{
+					command.CommandText = GetConfigTableVerificationQuery(request.DatabaseName, request.TableName, request.SchemaName);
+
+					command.ExecuteNonQuery();
+					using (IDataReader reader = command.ExecuteReader())
+					{
+						while (reader.Read())
+						{
+							return reader.GetInt32(0);
+						}
+					}
+					return 0;
+				}
+			});
+
+			return result == 1;
+		}
+
+		private TResult WithDbChange<T, TResult>(IDbConnection connection, T request, Func<T, TResult> operation) where T : ModelRequestBase
+		{
+			string currentDb = connection.Database;
+			if (connection.Database != request.DatabaseName)
+			{
+				connection.ChangeDatabase(request.DatabaseName);
+			}
+			TResult result = operation(request);
```

Comment:
You should set it back to the original in a finally block - otherwise an error in the operation will leave it in the wrong state

Reply:
good point. thanks

---
In reply to: [542822384](https://github.com/microsoft/sqltoolsservice/pull/1138#discussion_r542822384) [](ancestors = 542822384)